Feedback

NicMcPhee opened the pull request:
:wave:! GitHub Classroom created this pull request as a place for your teacher to leave feedback on your work. It will update automatically. **Don’t close or merge this pull request**, unless you’re instructed to do so by your teacher.

In this pull request, your teacher can leave comments and feedback on your code. Click the **Subscribe** button to be notified if that happens.

Click the **Files changed** or **Commits** tab to see all of the changes pushed to `main` since the assignment started. Your teacher can see this too.

<details>
<summary>
<strong>Notes for teachers</strong>
</summary>

Use this PR to leave feedback. Here are some tips:

- Click the **Files changed** tab to see all of the changes pushed to `main` since the assignment started. To leave comments on specific lines of code, put your cursor over a line of code and click the blue **+** (plus sign). To learn more about comments, read “[Commenting on a pull request](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/commenting-on-a-pull-request)”.
- Click the **Commits** tab to see the commits pushed to `main`. Click a commit to see specific changes.
- If you turned on autograding, then click the **Checks** tab to see the results.
- This page is an overview. It shows commits, line comments, and general comments. You can leave a general comment below.

For more information about this pull request, read “[Leaving assignment feedback in GitHub](https://classroom.github.com/help/leaving-feedback-in-github)”.
</details>


Subscribed: @schu4276

NicMcPhee committed:
Setting up GitHub Classroom Feedback

schu4276 committed:
I added the bats submodules needed to make bats work

schu4276 committed:
I copy and pasted the scripts from my other repository.

schu4276 committed:
adding this again

NicMcPhee commented on compiling/extract_and_compile.sh:
```diff
@@ -0,0 +1,15 @@
+# the directory specified by the user
+tar -xzf NthPrime.tgz 
+# Going into the NthPrimeDirectory that the file was extracted to
+#and compiling the C program that was extracted, generating a executable called NthPrime
```
Probably want a space after the `#`.

NicMcPhee commented on cleaning/big_clean.sh:
```diff
@@ -0,0 +1,23 @@
+# second script of lab 0
+##### CONSTANTS
+ARCHIVE_NAME=$1
+BASE=$(basename -s .tgz "$ARCHIVE_NAME")
```
To someone not familiar with the lab, it's not obvious what `BASE` is or what it's used for, especially since you don't use it until near the end of the script. You could either comment it here, or move it closer to where it's used, or both?

NicMcPhee replied:
Nice. 💯

NicMcPhee commented on cleaning/big_clean.sh:
```diff
@@ -0,0 +1,23 @@
+# remove all the files that contain "DELETE ME"
+# the -l flag displays the filenames that contain the delete me phrase 
+grep -Rl "DELETE ME" . | xargs rm   
+# Creating a new compressed tar archive that contains the files left over in the scratch di
```
It feels like this comment line got cut off at the end?

NicMcPhee commented on cleaning/big_clean.sh:
```diff
@@ -0,0 +1,23 @@
+tar -zcf cleaned_"$ARCHIVE_NAME" "$BASE"
+mv cleaned_"$ARCHIVE_NAME" "$HERE"
+cd "$HERE" || exit 
+#delete the scratch dir
```
You say you delete the scratch directory, but you don't actually seem to do that (and you should).

NicMcPhee reviewed (commented):
Nice work! I made a few specific notes (see below). The only "correctness" issue is that you don't appear to delete the scratch directory in `cleaning`, which you really should.

schu4276 commented:
Thanks for the feedback!

On Thu, Aug 27, 2020 at 5:52 PM, Nic McPhee <notifications@github.com>
wrote:

> *@NicMcPhee* commented on this pull request.
>
> Nice work. I made a few specific notes (see below). The only "correctness"
> issue is that you don't appear to delete the scratch directory in cleaning,
> which you really should.
> ------------------------------
>
> In compiling/extract_and_compile.sh
> <https://github.com/umm-csci-3403-fall-2020/lab-0-command-line-introduction-schu4276/pull/1#discussion_r478736344>
> :
>
> > @@ -0,0 +1,15 @@
> +#!/bin/bash
> +#first script of lab 0
> +##### CONSTANTS	
> +NUMBER=$1
> +##### MAIN
> +# extracting the contents of the tar archive, and putting them into
> +# the directory specified by the user
> +tar -xzf NthPrime.tgz
> +# Going into the NthPrimeDirectory that the file was extracted to
> +#and compiling the C program that was extracted, generating a executable called NthPrime
>
> Probably want a space after the #.
> ------------------------------
>
> In cleaning/big_clean.sh
> <https://github.com/umm-csci-3403-fall-2020/lab-0-command-line-introduction-schu4276/pull/1#discussion_r478737399>
> :
>
> > @@ -0,0 +1,23 @@
> +#!/bin/bash
> +# second script of lab 0
> +##### CONSTANTS
> +ARCHIVE_NAME=$1
> +BASE=$(basename -s .tgz "$ARCHIVE_NAME")
>
> To someone not familiar with the lab, it's not obvious what BASE is or
> what it's used for, especially since you don't use it until near the end of
> the script. You could either comment it here, or move it closer to where
> it's used, or both?
> ------------------------------
>
> In cleaning/big_clean.sh
> <https://github.com/umm-csci-3403-fall-2020/lab-0-command-line-introduction-schu4276/pull/1#discussion_r478737525>
> :
>
> > +##### CONSTANTS
> +ARCHIVE_NAME=$1
> +BASE=$(basename -s .tgz "$ARCHIVE_NAME")
> +#### MAIN
> +SCRATCH=$(mktemp -d)
> +# extracting the contents of the tar archive, and putting them into
> +# the new scratch dir
> +tar -xzf "$ARCHIVE_NAME" -C "$SCRATCH"
> +# perserving the current position so I can come back later
> +HERE=$(pwd)
> +# moving to the scratch dir
> +cd "$SCRATCH" || exit
> +# remove all the files that contain "DELETE ME"
> +# the -l flag displays the filenames that contain the delete me phrase
> +grep -Rl "DELETE ME" . | xargs rm
> +# Creating a new compressed tar archive that contains the files left over in the scratch di
>
> It feels like this comment line got cut off at the end?
> ------------------------------
>
> In cleaning/big_clean.sh
> <https://github.com/umm-csci-3403-fall-2020/lab-0-command-line-introduction-schu4276/pull/1#discussion_r478737707>
> :
>
> > +SCRATCH=$(mktemp -d)
> +# extracting the contents of the tar archive, and putting them into
> +# the new scratch dir
> +tar -xzf "$ARCHIVE_NAME" -C "$SCRATCH"
> +# perserving the current position so I can come back later
> +HERE=$(pwd)
> +# moving to the scratch dir
> +cd "$SCRATCH" || exit
> +# remove all the files that contain "DELETE ME"
> +# the -l flag displays the filenames that contain the delete me phrase
> +grep -Rl "DELETE ME" . | xargs rm
> +# Creating a new compressed tar archive that contains the files left over in the scratch di
> +tar -zcf cleaned_"$ARCHIVE_NAME" "$BASE"
> +mv cleaned_"$ARCHIVE_NAME" "$HERE"
> +cd "$HERE" || exit
> +#delete the scratch dir
>
> You say you delete the scratch directory, but you don't actually seem to
> do that (and you should).
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/umm-csci-3403-fall-2020/lab-0-command-line-introduction-schu4276/pull/1#pullrequestreview-477134014>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AQVAZXAXB45TPHCLD7GL3TDSC3PTNANCNFSM4QGEMSXQ>
> .
>

schu4276 committed:
I have made some additional comments for clarity as per feedback from nic
I also added to the script so that the scratch directory is deleted at the end of the script

schu4276 committed:
I added a space to one of the comments as per feedback from nic

NicMcPhee reviewed (commented):
No problem – I'm glad to have helped.

Nice changes – I approve! 👍

UtkarshKr007 reviewed (approved):
Even if I didn't know Bash Script, I could follow what each line of code is doing by your comments, awesome! I like how you explained what certain flags do as well.
